be/src/exec/exec_node.cpp

```diff
@@ -672,13 +675,51 @@ void ExecNode::eval_conjuncts(const std::vector<ExprContext*>& ctxs, vectorized:
 void ExecNode::eval_join_runtime_filters(vectorized::Chunk* chunk) {
     if (chunk == nullptr) return;
     _runtime_filter_collector.evaluate(chunk);
+    eval_filter_null_values(chunk);
 }
 
 void ExecNode::eval_join_runtime_filters(vectorized::ChunkPtr* chunk) {
     if (chunk == nullptr) return;
     eval_join_runtime_filters(chunk->get());
 }
 
+void ExecNode::eval_filter_null_values(vectorized::Chunk* chunk) {
+    if (_filter_null_value_columns.size() == 0) return;
+    size_t before_size = chunk->num_rows();
+    if (before_size == 0) return;
+
+    // lazy allocation.
+    vectorized::Buffer<uint8_t> selection(0);
+
+    for (SlotId slot_id : _filter_null_value_columns) {
+        const ColumnPtr& c = chunk->get_column_by_slot_id(slot_id);
+        const vectorized::NullableColumn* nullable_column =
+                vectorized::ColumnHelper::as_raw_column<vectorized::NullableColumn>(c);
+        if (!nullable_column->has_null()) continue;
+        if (selection.size() == 0) {
+            selection.assign(before_size, 1);
+        }
+        // how many data() should we call? I really don't know.
+        // let compiler tells me.
+        // let compiler does vectorization.
+        // let compiler does everything.
+        // let's pray for compiler,
+        // till the end of the world.
+        const uint8_t* bits = nullable_column->null_column()->raw_data();
+        uint8_t* sel = selection.data();
+        for (size_t i = 0; i < before_size; i++) {
+            sel[i] &= (1 - bits[i]);
```

Comment:
Why not !bit[i]?

Reply:
Maybe  `(1-bits[i])` is more efficient. Let's see asm code

-----
The following code has two SIMD insts:
1. pcmpeqb
2. pand

```
void neg(uint8_t* data, size_t size) {
    for (size_t i=0;i<size;i++) {
        data[i]=!data[i];
    }
}
```

![image](https://user-images.githubusercontent.com/1081215/142981305-5e2aa7d5-8c22-4aa1-847b-bf05ee964922.png)


---
The following code has one SIMD inst:
- psubb

```
void neg2(uint8_t* data, size_t size) {
    for (size_t i=0;i<size;i++) {
        data[i]=1-data[i];
    }
}


```

![image](https://user-images.githubusercontent.com/1081215/142981535-914cecd1-18b2-46dd-94e7-8203922c3b46.png)

Reply:
Oh, there is another `movdqa`.  It should be the same.